src/dec128.rs

```diff
@@ -76,6 +78,114 @@ impl Hash for d128 {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for d128{
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_struct("d128", d128MapVisitor {
+            value: self,
+            state: 0,
+        })
+    }
+}
+
+#[cfg(feature = "serde")]
+#[allow(non_camel_case_types)]
+struct d128MapVisitor<'a> {
+    value: &'a d128,
+    state: u8
+}
+
+#[cfg(feature = "serde")]
+impl<'a> ser::MapVisitor for d128MapVisitor<'a> {
+    fn visit<S>(&mut self, serializer: &mut S) -> Result<Option<()>, S::Error>
+        where S: Serializer
+    {
+        match self.state {
+            0 => {
+                self.state += 1;
+                Ok(Some(try!(serializer.serialize_struct_elt("bytes", &self.value.bytes))))
```

Comment:
Should we be outputing bytes? In rustc_serialize I output the decimal representation. This is because it is lossless and easier to read. I think we should be consistent at the very least.